kotest-assertions/kotest-assertions-core/src/commonMain/kotlin/io/kotest/matchers/sequences/matchers.kt

```diff
@@ -37,12 +37,14 @@ fun <T> containOnlyNulls() = object : Matcher<Sequence<T>> {
 fun <T> Sequence<T>.shouldContainNull() = this should containNull()
 fun <T> Sequence<T>.shouldNotContainNull() = this shouldNot containNull()
 fun <T> containNull() = object : Matcher<Sequence<T>> {
-   override fun test(value: Sequence<T>) =
-      MatcherResult(
-         value.any { it == null },
+   override fun test(value: Sequence<T>): MatcherResult {
+      val indexOfFirstNull = value.mapIndexed { index, t -> index to t }.firstOrNull { it.second == null }
```

Comment:
🟡 We can cleanup this guy with `indexOfFirst`

```suggestion
      val indexOfFirstNull = value.indexOfFirst { it == null }
```

Reply:
oops forgot about that thing, TYM